.github/workflows/continuous-integration.yml

```diff
@@ -580,6 +641,209 @@ jobs:
           path: cypress/results
           retention-days: 1
 
+  stress-test-cypress-tests:
+    name: Stress Test Cypress Tests
+    runs-on: self-hosted
+    timeout-minutes: 60
+    needs: [build, cypress-tests-prep]
+    if: |
+      needs.build.result == 'success' &&
+      needs.cypress-tests-prep.result == 'success' &&
+      needs.cypress-tests-prep.outputs.tests-to-stress-test != '[]'
+    container:
+      image: public.ecr.aws/cypress-io/cypress/browsers:node16.13.2-chrome100-ff98
+      options: -u 1001:1001 -v /usr/local/share:/share
+
+    env:
+      CHROMEDRIVER_FILEPATH: /share/chrome_driver/chromedriver
+      NODE_EXTRA_CA_CERTS: /share/ca-certificates/VA-Internal-S2-RCA1-v1.cer.crt
+      CI_NODE_INDEX: ${{ needs.cypress-tests-prep.outputs.ci_node_index }}
+
+    steps:
+      # The following statement is included in each step because of a bug in
+      # GitHub's branch protection:
+      #
+      # if: needs.cypress-tests-prep.outputs.tests != '[]'
+      #
+      # Previously, if no tests were selected, the branch protection check that
+      # requires the cypress-tests job to run was not satisfied. This update forces
+      # the job to always run, and skips each step if no tests are selected.
+      # Previously, the above conditional was included in the job's if statement.
+      - name: Configure AWS credentials
+        if: needs.cypress-tests-prep.outputs.tests-to-stress-test != '[]'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-gov-west-1
+
+      - name: Checkout vets-website
+        if: needs.cypress-tests-prep.outputs.tests-to-stress-test != '[]'
+        uses: actions/checkout@v3
+
+      - name: Download production build artifact
+        if: needs.cypress-tests-prep.outputs.tests-to-stress-test != '[]'
+        uses: actions/download-artifact@v3
+        with:
+          name: vagovprod.tar.bz2
+
+      - name: Unpack build
+        if: needs.cypress-tests-prep.outputs.tests-to-stress-test != '[]'
+        run: |
+          mkdir -p build/vagovprod
+          tar -C build/vagovprod -xjf vagovprod.tar.bz2
+
+      - name: Install dependencies
+        if: needs.cypress-tests-prep.outputs.tests-to-stress-test != '[]'
+        uses: ./.github/workflows/install
+        timeout-minutes: 20
+        with:
+          key: ${{ hashFiles('yarn.lock') }}
+          yarn_cache_folder: .cache/yarn
+          path: |
+            .cache/yarn
+            /github/home/.cache/Cypress
+            node_modules
+
+      - name: Start server
+        if: needs.cypress-tests-prep.outputs.tests-to-stress-test != '[]'
+        run: node src/platform/testing/e2e/test-server.js --buildtype=vagovprod --port=3001 &
+
+      - name: Run Cypress tests
```

Comment:
Yes, that's correct. Test selection still works the same, with one twist: if a selected test is a new or changed test, it will run in the `stress-test-cypress-tests` job.

Reply:
Thanks for confirming! So in theory, the `stress-test-cypress-tests` job shouldn't take longer than the `cypress-tests` job since it will only stress test new or updated tests in CI? 

One instance where I could see `stress-test-cypress-tests` taking longer than `cypress-tests` is in isolated app builds for a newer app that's beginning to get Cypress tests added to it. If Cypress tests are being added to an app for the first time, both `cypress-tests` and `stress-test-cypress-tests` would run the same number of tests, but the stress test would run those tests multiple times, resulting in it taking longer than the `cypress-test` job. Would this be the case?

Reply:
If it's a new test, it won't run in `cypress-tests`, it will only run in `stress-test-cypress-tests`. Only tests that don't need to be stress-tested run in `cypress-tests`.

So if the branch is for a new app with e2e tests, all tests will only run in `stress-test-cypress-tests`. This is to ensure that flaky tests (or tests that fail even 1x when they are run in a loop 25x) are never merged to master. So, yes, `stress-test-cypress-tests` will take longer to run in this case.

I hope that makes sense :-)

Reply:
Cool, that makes sense. So new tests and updated flaky tests will be run in a loop up to 25 times if they don’t fail sooner. Have there been workflow runs that test how long it takes to stress test a handful of tests that reach 25 iterations in CI? It seems like stress testing 1 or 2 tests in a loop that makes it up to 25 iterations may not take that long, but if the number of tests triples or quadruples, that could add some time to the stress test job in CI